src/pages/login/components/signup-form.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useState } from 'react';
+
+import { Button } from 'common/buttons/button';
+import { isValidPassword, isValidEmail, checkSignupInputs } from 'helpers/validators';
+import { classnames } from 'helpers/utils';
+import { User } from 'networking/types/user';
+import { goToPage, RouteName } from 'routes';
+import styles from './card.module.scss';
+
+const DEFAULT_REMINDER = 'Password must be 8 characters long and include special characters';
+
+const SignUpForm = () => {
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [passwordConfirmation, setPasswordConfirmation] = useState('');
+  const [reminder, setReminder] = useState(DEFAULT_REMINDER);
+  const [emailIsValid, setEmailIsValid] = useState(true);
+  const [passwordIsValid, setPasswordIsValid] = useState(true);
+  const [confirmationIsValid, setConfirmationIsValid] = useState(true);
+  const [error, setError] = useState(false);
```

Comment:
This is a lot of state, which is completely fine, but it makes me appreciate libraries that handle forms more :smile:

Reply:
Another approach you could take is keep all of the form input values inside of an object like:
`const [formInputs, setFormInputs] = {firstName: '', lastName: '', email: '', ...}`

Then it becomes easier to have one generic handler `handleInputChange` which updates the state looking at the input's name